app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt

```diff
@@ -13,25 +13,21 @@ import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
 import com.example.randomusergenerator.R
 import com.example.randomusergenerator.data.local.UserViewState
 import com.example.randomusergenerator.ui.resources.Dimensions.space_x2
 import com.example.randomusergenerator.user.viewmodel.UserViewModel
 
 @Composable
-fun UserScreen() {
-    val userViewModel: UserViewModel = viewModel()
+fun UserScreen(userViewModel: UserViewModel, navController: NavController) {
     val userViewState by userViewModel.uiState.collectAsState()
 
-    LaunchedEffect(key1 = true) {
-        userViewModel.getAllUsers()
```

Comment:
Why did you remove this?

Reply:
I wanted to limit the network calls made. It always called get users every time the user comes to the first screen

Reply:
Kindly add a back icon to the detail page.